service/src/main/java/uk/gov/hmcts/reform/fpl/service/PlacementService.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.Child;
+import uk.gov.hmcts.reform.fpl.model.Placement;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
+import uk.gov.hmcts.reform.fpl.utils.ElementUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Function;
+
+import static java.util.Optional.ofNullable;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.element;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.findElement;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class PlacementService {
+
+    public boolean hasSingleChild(CaseData caseData) {
+        return caseData.getAllChildren().size() == 1;
+    }
+
+    public Element<Child> getChild(CaseData caseData, UUID childId) {
+        return findElement(childId, caseData.getAllChildren()).orElse(null);
+    }
+
+    public DynamicList getChildrenList(CaseData caseData, Element<Child> selectedChild) {
+        List<Element<Child>> children = caseData.getAllChildren();
+        UUID selectedChildId = ofNullable(selectedChild).map(Element::getId).orElse(null);
+        Function<Child, String> labelProducer = child -> child.getParty().getFullName();
+
+        return ElementUtils.asDynamicList(children, selectedChildId, labelProducer);
+    }
+
+    public Placement getPlacement(CaseData caseData, Element<Child> child) {
+        return findPlacement(caseData.getPlacements(), child.getId())
+            .map(Element::getValue)
+            .orElse(Placement.builder()
+                .childId(child.getId())
+                .childName(child.getValue().getParty().getFullName())
+                .build());
+    }
+
+    public List<Element<Placement>> setPlacement(CaseData caseData, Placement placement) {
+        List<Element<Placement>> placements = new ArrayList<>(caseData.getPlacements());
+
+        Optional<Element<Placement>> existingPlacement = findPlacement(placements, placement.getChildId());
+
+        if (existingPlacement.isPresent()) {
```

Comment:
How about `ifPresentOrElse`?

Reply:
done,

but  I dont think that having 2 lambdas inside method invocation is most readable solution. What i like about this is that optional is already unboxed and ready to use